lib/marloss/store.rb

```diff
@@ -13,35 +13,61 @@ def initialize(table, hash_key, ttl: 30, client_options: {})
     end
 
     def create_table
-      client.create_table(
-        attribute_definitions: [
-          {
-            attribute_name: hash_key,
-            attribute_type: "S",
-          }
-        ],
-        key_schema: [
-          {
-            attribute_name: hash_key,
-            key_type: "HASH",
-          }
-        ],
-        provisioned_throughput: {
-          read_capacity_units: 5,
-          write_capacity_units: 5,
-        },
-        table_name: table
-      )
+      begin
```

Comment:
this method is becoming quite big 😬
can you split it into 3 smaller private methods that then gets called by the `create_table` method?

something like:

```ruby
def create_table
  create_ddb_table
  wait_until_ddb_table_exists
  set_ddb_table_ttl
end

private def create_ddb_table
  client.create_table(
    ...
  )
rescue Aws::DynamoDB::Errors::ResourceInUseException => e
  case e.massage
  when "Table already exists: #{table}"
    Marloss.logger.warn("DynamoDB table #{table} already exists")
  else
    raise
  end
end

private def wait_until_ddb_table_exists
  client.wait_until(:table_exists, table_name: table) do |w|
    w.max_attempts = 10
    w.delay = 1
  end
rescue Aws::Waiters::Errors::WaiterFailed => e
  Marloss.logger.error("Failed waiting for initialization of table #{table}")
  raise
end

private def set_ddb_table_ttl
  client.update_time_to_live(
    ...
  )
rescue Aws::DynamoDB::Errors::ValidationException => e
  case e.message
  when "TimeToLive is already enabled"
    Marloss.logger.warn("TTL attribute is already configured for table #{table}")
  else
    raise
  end
end
```

Reply:
Easy!

Reply:
that could cause confusion whether the TTL is required or not. I think exposing a simple method would create a better user experience

Reply:
sounds fair. I've added separate wait function